File: NWorldPermissions/src/top/nololiyt/worldpermissions/commands/marks/MarksRouter.java
package top.nololiyt.worldpermissions.commands.marks;

import top.nololiyt.worldpermissions.commands.CommandLayer;
import top.nololiyt.worldpermissions.commands.Router;

import java.util.HashMap;
import java.util.Map;

public class MarksRouter extends Router
{
    protected final static String layerName = "marks";
    
    @Override
    public String permissionName()
    {
        return layerName;
    }
    
    @Override
    public String messageKey()
    {
        return layerName;
    }
    
    private Map<String, CommandLayer> commandLayers = new HashMap<String, CommandLayer>()
    {
        {
            put("add", new AddExecutor());
            put("remove", new RemoveExecutor());
            put("list", new ListExecutor());
        }
    };
    
    @Override
    protected Map<String, CommandLayer> nextLayers()
    {
        return commandLayers;
    }
}

File: NWorldPermissions/src/top/nololiyt/worldpermissions/commands/marks/MarksRouter.kt
package top.nololiyt.worldpermissions.commands.marks

import top.nololiyt.worldpermissions.commands.CommandLayer
import top.nololiyt.worldpermissions.commands.Router

class MarksRouter : Router()
{
    override fun permissionName(): String?
    {
        return layerName;
    }
    
    override fun messageKey(): String?
    {
        return layerName;
    }
    
    /**
     * Return the next layer matching the arg.
     * If no layer match it, please return 'null' and the help list will be sent.
     *
     * @param arg
     * @return
     */
    override fun nextLayer(arg: String): CommandLayer?
    {
        return when (arg)
        {
            "add"    -> AddExecutor()
            "remove" -> RemoveExecutor()
            else     -> null
        };
    }
    
    companion object
    {
        private const val layerName= "marks";
    }
}